Plugins/Player/Player.cpp

```diff
@@ -1336,4 +1344,163 @@ ArgumentStack Player::SetResManOverride(ArgumentStack&& args)
     return Services::Events::Arguments();
 }
 
+ArgumentStack Player::AddCustomJournalEntry(ArgumentStack&& args)
+{
+    int32_t retval = -1;
+    if (auto *pPlayer = player(args))
+    {
+        auto *pCreature = Globals::AppManager()->m_pServerExoApp->GetCreatureByGameObjectID(pPlayer->m_oidNWSObject);
+        
+        if (pCreature && pCreature->m_pJournal)
+        {
+            auto entries = pCreature->m_pJournal->m_lstEntries;
+            
+            const auto questName = Services::Events::ExtractArgument<std::string>(args);
+            const auto questText = Services::Events::ExtractArgument<std::string>(args);
+            const auto tag = Services::Events::ExtractArgument<std::string>(args);
+            
+            ASSERT_OR_THROW(tag.size() > 0);
+            
+            const auto state = Services::Events::ExtractArgument<int32_t>(args);
+            const auto priority = Services::Events::ExtractArgument<int32_t>(args);
+            const auto completed = Services::Events::ExtractArgument<int32_t>(args);
+            const auto diplayed = Services::Events::ExtractArgument<int32_t>(args);
+            const auto updated = Services::Events::ExtractArgument<int32_t>(args);
+            
+            auto calDay = Services::Events::ExtractArgument<int32_t>(args);
+            auto timeDay = Services::Events::ExtractArgument<int32_t>(args);
+            auto silentUpdate = Services::Events::ExtractArgument<int32_t>(args);
+            
+            ASSERT_OR_THROW(state >= 0);
+            ASSERT_OR_THROW(priority >= 0);
+            ASSERT_OR_THROW(completed >= 0);
+            ASSERT_OR_THROW(updated >= 0);
+            ASSERT_OR_THROW(silentUpdate >= 0);
+            
+            
+            //If server owner leaves this 0 - the entry will be added with todays date
+            if (calDay <= 0)
+            {
+                calDay = Globals::AppManager()->m_pServerExoApp->GetWorldTimer()->GetWorldTimeCalendarDay();
+            }
+            //If server owner leaves this 0 - the entry will be added with now() time
+            if(timeDay <= 0)
```

Comment:
A few generic style nits throughout the change:
- space between keyword and paren: `if (x)`, not `if(x)`.
- space around assignment operator `=` on both sides:  `a = x;` not `a=x`, `a= x`, `a =x`.
- Avoid superfluous newlines, such as empty lines after `{`